PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['hr.resume.line']
-    _inherit = ['survey.survey']
-    _inherit = ['survey.user_input']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills_survey/models/hr_resume_line.py
+++ b/addons/hr_skills_survey/models/hr_resume_line.py
@@ -6,7 +6,7 @@ from odoo import api, fields, models
 
 
 class HrResumeLine(models.Model):
-    _inherit = ['hr.resume.line']
+    _inherit = 'hr.resume.line'
 
     display_type = fields.Selection(selection_add=[('certification', 'Certification')])
     department_id = fields.Many2one(related="employee_id.department_id", store=True)

--- a/addons/hr_skills_survey/models/survey_survey.py
+++ b/addons/hr_skills_survey/models/survey_survey.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class SurveySurvey(models.Model):
-    _inherit = ['survey.survey']
+    _inherit = 'survey.survey'
 
     certification_validity_months = fields.Integer(
         'Validity', required=False,

--- a/addons/hr_skills_survey/models/survey_user.py
+++ b/addons/hr_skills_survey/models/survey_user.py
@@ -9,7 +9,7 @@ from odoo.tools import html2plaintext
 
 
 class SurveyUser_Input(models.Model):
-    _inherit = ['survey.user_input']
+    _inherit = 'survey.user_input'
 
     def _mark_done(self):
         """ Will add certification to employee's resume if
